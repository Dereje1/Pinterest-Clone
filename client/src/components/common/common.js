import React from 'react';
import PropTypes from 'prop-types';
import TwitterIcon from '@mui/icons-material/Twitter';
import GoogleIcon from '@mui/icons-material/Google';
import GitHubIcon from '@mui/icons-material/GitHub';
import DownloadDoneIcon from '@mui/icons-material/DownloadDone';
import './common.scss';

export const getProviderIcons = ({ fontSize }) => (
  {
    twitter: { icon: <TwitterIcon style={{ fontSize }} />, color: '#1DA1F2' },
    google: { icon: <GoogleIcon style={{ fontSize }} />, color: '#4285F4' },
    github: { icon: <GitHubIcon style={{ fontSize }} />, color: '#1d7b20' },
  }
);

export function Loading({ imagesLoaded, ready }) {
  return (
    <div id="loadingcontainer">
      {
        imagesLoaded && ready ? <DownloadDoneIcon />
          : (
            <>
              <div className="bubbles A" />
              <div className="bubbles B" />
              <div className="bubbles C" />
            </>
          )
      }
    </div>
  );
}

Loading.defaultProps = {
  imagesLoaded: undefined,
  ready: undefined,
};

Loading.propTypes = {
  imagesLoaded: PropTypes.bool,
  ready: PropTypes.bool,
};
