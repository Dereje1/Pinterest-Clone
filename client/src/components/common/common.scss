@import "../../keyframes.scss";

#loadingcontainer{
    display: flex;
    justify-content: space-evenly;
    align-items: center;
    width: 100px;
    height: 50px;
    margin-right: auto;
    margin-left: auto;
    .bubbles{
      background: #676b71;
      border-radius: 100%;
      margin: auto;
      @include animation('bubbleanimate');
      animation-duration: 1s;
      animation-iteration-count: infinite;
      animation-timing-function: ease-in-out;
      animation-direction: alternate;
    &.A{
      width: 20px;
      height: 20px;
    }
    &.B{
      width: 30px;
      height: 30px;
      animation-delay: .25s;
    }
    &.C{
      width: 40px;
      height: 40px;
      animation-delay: .5s;
    }
  }
  }

  @include keyframes(bubbleanimate){
    0%{
      transform: scale(0.2)
    }
    100%{
      transform: scale(1)
    }
  }