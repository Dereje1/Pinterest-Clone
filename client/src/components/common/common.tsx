import React from 'react';
import TwitterIcon from '@mui/icons-material/Twitter';
import GoogleIcon from '@mui/icons-material/Google';
import GitHubIcon from '@mui/icons-material/GitHub';
import DownloadDoneIcon from '@mui/icons-material/DownloadDone';
import Button from '@mui/material/Button';
import ButtonGroup from '@mui/material/ButtonGroup';
import { providerIconsType } from '../../interfaces';
import './common.scss';

export const getProviderIcons = ({ fontSize }: {fontSize: number}): providerIconsType => (
  {
    twitter: { icon: <TwitterIcon style={{ fontSize }} />, color: '#1DA1F2' },
    google: { icon: <GoogleIcon style={{ fontSize }} />, color: '#4285F4' },
    github: { icon: <GitHubIcon style={{ fontSize }} />, color: '#1d7b20' },
  }
);

interface LoadingProps {
  imagesLoaded?: boolean,
  ready?: boolean
}

export function Loading({ imagesLoaded, ready }: LoadingProps) {
  return (
    <div id="loadingcontainer">
      {
        imagesLoaded && ready ? <DownloadDoneIcon />
          : (
            <>
              <div className="bubbles A" />
              <div className="bubbles B" />
              <div className="bubbles C" />
            </>
          )
      }
    </div>
  );
}
Loading.defaultProps = {
  imagesLoaded: undefined,
  ready: undefined,
};

interface UserPinsSelectorProps {
  displaySetting: string,
  setDisplaySetting: (arg: string) => void,
}

export function UserPinsSelector({ displaySetting, setDisplaySetting }: UserPinsSelectorProps) {
  return (
    <ButtonGroup variant="text" aria-label="text button group" sx={{ mt: 3 }}>
      <Button
        color={displaySetting === 'created' ? 'secondary' : 'primary'}
        onClick={() => setDisplaySetting('created')}
      >
        Pins created
      </Button>
      <Button
        color={displaySetting === 'saved' ? 'secondary' : 'primary'}
        onClick={() => setDisplaySetting('saved')}
      >
        Pins saved
      </Button>
    </ButtonGroup>
  );
}
