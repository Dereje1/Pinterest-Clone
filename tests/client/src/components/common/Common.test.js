import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Loading, UserPinsSelector } from '../../../../../client/src/components/common/common.tsx';

describe('Loading...', () => {
  test('Will render the loading bubbles', () => {
    const wrapper = shallow(<Loading />);
    const bubbles = wrapper.find({ id: 'loadingcontainer' });
    expect(toJson(bubbles)).toMatchSnapshot();
  });

  test('Will not render the loading bubbles', () => {
    const wrapper = shallow(<Loading ready imagesLoaded />);
    const bubbles = wrapper.find({ id: 'loadingcontainer' });
    expect(toJson(bubbles)).toMatchSnapshot();
  });
});

describe('The user pins selector', () => {
  test('Will highlight the owned button', () => {
    const setDisplaySetting = jest.fn();
    const wrapper = shallow(<UserPinsSelector setDisplaySetting={setDisplaySetting} displaySetting="created" />);
    // for owned button
    const ownedButton = wrapper.find('ForwardRef(Button)').at(0);
    const savedButton = wrapper.find('ForwardRef(Button)').at(1);
    expect(ownedButton.props().color).toBe('secondary');
    expect(savedButton.props().color).toBe('primary');
    savedButton.props().onClick();
    expect(setDisplaySetting).toHaveBeenCalledWith('saved');
  });
  test('Will highlight the saved button', () => {
    const setDisplaySetting = jest.fn();
    const wrapper = shallow(<UserPinsSelector setDisplaySetting={setDisplaySetting} displaySetting="saved" />);
    // for owned button
    const ownedButton = wrapper.find('ForwardRef(Button)').at(0);
    const savedButton = wrapper.find('ForwardRef(Button)').at(1);
    expect(ownedButton.props().color).toBe('primary');
    expect(savedButton.props().color).toBe('secondary');
    ownedButton.props().onClick();
    expect(setDisplaySetting).toHaveBeenCalledWith('created');
  });
});
