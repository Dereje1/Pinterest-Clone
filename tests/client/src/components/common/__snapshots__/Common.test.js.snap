// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading... Will not render the loading bubbles 1`] = `
<div
  id="loadingcontainer"
>
  <Memo(ForwardRef(DownloadDoneIcon)) />
</div>
`;

exports[`Loading... Will render the loading bubbles 1`] = `
<div
  id="loadingcontainer"
>
  <div
    className="bubbles A"
  />
  <div
    className="bubbles B"
  />
  <div
    className="bubbles C"
  />
</div>
`;
